src/core/qgsmaplayerutils.cpp

```diff
@@ -138,7 +138,7 @@ bool QgsMapLayerUtils::updateLayerSourcePath( QgsMapLayer *layer, const QString
   parts.insert( QStringLiteral( "path" ), newPath );
   const QString newUri = QgsProviderRegistry::instance()->encodeUri( layer->providerType(), parts );
   layer->setDataSource( newUri, layer->name(), layer->providerType() );
-  return true;
+  return layer->isValid();
```

Comment:
I don't think this change is correct. From the docs: "Returns TRUE if the layer was updated, or FALSE if the layer was not updated". Even if the layer is not valid after changing the path the path still HAS been updated, so we should return true here.

Reply:
There's many uses -- eg setting a layer path to a temporarily unavailable source but which you know will be available on other installations. (such as a client's network which you can't access).

SLYR also uses this extensively to set layer paths to the original paths as extracted from an MXD/APRX. It's then up to the user to use all the tools QGIS has available for fixing bad layer paths manually.

Reply:
Ok I understand better now. 

Should I keep the other changes (file copies in the tests)?

Reply:
When updating the layer path to non existent files, the gdal provider generates errors leading to confusion when reading the logs (IMHO).

Reply:
I wouldn't worry too much about that -- there's similar errors thrown all thoughout the test suite as we deliberately try to open non existing files.

Reply:
ok so I close